File: Facets/OZLAppDelegate.m
//
//  OZLAppDelegate.m
//  RedmineMobile
//
//  Created by Lee Zhijie on 7/14/13.

#import "OZLAppDelegate.h"
#import "OZLIssueListViewController.h"
#import "OZLSingleton.h"
#import "OZLModelProject.h"
#import "OZLAccountViewController.h"
#import "OZLMainTabControllerViewController.h"
#import "OZLURLProtocol.h"
#import "OZLNetwork.h"

#import <HockeySDK/HockeySDK.h>

@implementation OZLAppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    
#ifndef DEBUG
    [[BITHockeyManager sharedHockeyManager] configureWithIdentifier:@"8d240e4921f15253d040d9347ad7d9ac"];
    [[BITHockeyManager sharedHockeyManager] startManager];
    [[[BITHockeyManager sharedHockeyManager] authenticator] authenticateInstallation];
#endif // DEBUG
    
    [OZLNetwork sharedInstance];
    [OZLSingleton sharedInstance];
    
    [NSURLProtocol registerClass:[OZLURLProtocol class]];
    
    self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
    self.window.tintColor = [UIColor facetsBrandColor];
    self.window.backgroundColor = [UIColor whiteColor];

    OZLMainTabControllerViewController *mainVC = [[OZLMainTabControllerViewController alloc] init];
    self.window.rootViewController = mainVC;
    [self.window makeKeyAndVisible];
    
    return YES;
}

@end

File: Facets/OZLAppDelegate.swift
//
//  OZLAppDelegate.swift
//  Facets
//
//  Created by Justin Hill on 1/1/16.
//  Copyright © 2016 Justin Hill. All rights reserved.
//

import UIKit
import HockeySDK

class OZLAppDelegate: UIResponder, UIApplicationDelegate {
    var window: UIWindow?

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
        if OZLThirdPartyIntegrations.Enabled {
            #if !DEBUG
                BITHockeyManager.shared().configure(withIdentifier: OZLThirdPartyIntegrations.HockeyApp.AppKey)
                BITHockeyManager.shared().start()
                BITHockeyManager.shared().authenticator.authenticateInstallation()
            #endif

            HelpshiftCore.initialize(with: HelpshiftAll.sharedInstance())
            HelpshiftCore.install(forApiKey: OZLThirdPartyIntegrations.Helpshift.APIKey,
                domainName: OZLThirdPartyIntegrations.Helpshift.Domain,
                appID: OZLThirdPartyIntegrations.Helpshift.AppId
            )
        }

        OZLNetwork.sharedInstance()
        OZLSingleton.sharedInstance()

        URLProtocol.registerClass(OZLURLProtocol.self)

        self.window = UIWindow(frame: UIScreen.main.bounds)
        self.window?.tintColor = UIColor.facetsBrand()
        self.window?.backgroundColor = UIColor.white

        let vc = OZLMainTabControllerViewController()
        self.window?.rootViewController = vc
        self.window?.makeKeyAndVisible()

        return true
    }

    func applicationDidBecomeActive(_ application: UIApplication) {
        OZLSingleton.sharedInstance().startSessionUpkeep()
    }

    func applicationDidEnterBackground(_ application: UIApplication) {
        OZLSingleton.sharedInstance().suspendSessionUpkeep()
    }
}
